Honeycomb Bravo.lua

```diff
@@ -116,396 +133,617 @@ function array_has_true(array)
-				set_led(LED.FCU_IAS, true)
+		-- VS
+		if PROFILE == "Toliss/32x" then
+			if vs[0] == 107 and (int_to_bool(ap1[0]) or int_to_bool(ap2[0])) then
+				set_led(LED.FCU_VS, true)
 			else
-				set_led(LED.FCU_IAS, get_ap_state(ias))
+				set_led(LED.FCU_VS, false)
 			end
+		else
+			set_led(LED.FCU_VS, get_ap_state(vs))
+		end
 
-			-- AUTOPILOT
+		-- IAS
+		if bitwise.band(autopilot_state[0], 8) > 0 then
+			-- Speed-by-pitch Engage AKA Flight Level Change
+			-- See "Aliasing of Flight-Level Change With Speed Change" on https://developer.x-plane.com/article/accessing-the-x-plane-autopilot-from-datarefs/
+			set_led(LED.FCU_IAS, true)
+		else
+			set_led(LED.FCU_IAS, get_ap_state(ias))
+		end
+
+		-- AUTOPILOT
+		if PROFILE == "Toliss/32x" then
+			set_led(LED.FCU_AP, int_to_bool(ap1[0]) or int_to_bool(ap2[0]))
+		else
 			set_led(LED.FCU_AP, int_to_bool(ap[0]))
+		end
 
-			-- Landing gear
-			local gear_leds = {}
+		-- Landing gear
+		local gear_leds = {}
 
-			for i = 1, 3 do
-				gear_leds[i] = {nil, nil} -- green, red
+		for i = 1, 3 do
+			gear_leds[i] = { nil, nil } -- green, red
 
-				if retractable_gear[0] == 0 then
-					-- No retractable landing gear
+			if retractable_gear[0] == 0 then
+				-- No retractable landing gear
+			else
+				if gear[i - 1] == 0 then
+					-- Gear stowed
+					gear_leds[i][1] = false
+					gear_leds[i][2] = false
+				elseif gear[i - 1] == 1 then
+					-- Gear deployed
+					gear_leds[i][1] = true
+					gear_leds[i][2] = false
 				else
-					if gear[i - 1] == 0 then
-						-- Gear stowed
-						gear_leds[i][1] = false
-						gear_leds[i][2] = false
-					elseif gear[i - 1] == 1 then
-						-- Gear deployed
-						gear_leds[i][1] = true
-						gear_leds[i][2] = false
-					else
-						-- Gear moving
-						gear_leds[i][1] = false
-						gear_leds[i][2] = true
-					end
+					-- Gear moving
+					gear_leds[i][1] = false
+					gear_leds[i][2] = true
 				end
 			end
+		end
 
-			set_led(LED.LDG_N_GREEN, gear_leds[1][1])
-			set_led(LED.LDG_N_RED, gear_leds[1][2])
-			set_led(LED.LDG_L_GREEN, gear_leds[2][1])
-			set_led(LED.LDG_L_RED, gear_leds[2][2])
-			set_led(LED.LDG_R_GREEN, gear_leds[3][1])
-			set_led(LED.LDG_R_RED, gear_leds[3][2])
-
-			-- MASTER WARNING
-			set_led(LED.ANC_MSTR_WARNG, int_to_bool(master_warn[0]))
-
-			-- ENGINE FIRE
+		set_led(LED.LDG_N_GREEN, gear_leds[1][1])
+		set_led(LED.LDG_N_RED, gear_leds[1][2])
+		set_led(LED.LDG_L_GREEN, gear_leds[2][1])
+		set_led(LED.LDG_L_RED, gear_leds[2][2])
+		set_led(LED.LDG_R_GREEN, gear_leds[3][1])
+		set_led(LED.LDG_R_RED, gear_leds[3][2])
+
+		-- MASTER WARNING
+		set_led(LED.ANC_MSTR_WARNG, int_to_bool(master_warn[0]))
+
+		-- ENGINE/APU FIRE
+		if PROFILE == 'Toliss/32x' then
+			my_fire = false
+			if apu_fire[20] > 0 then
+				my_fire = true
+			end
+			for i = 11, 17 do
+				if i == 11 or i == 13 or i == 15 or i == 17 then
```

Comment:
Hah this is so wacky. Do we have to skip the other entries in the `eng_fire` table because they contain bad data we have to ignore?